test/functional/test_2_linkclicker.py

```diff
@@ -0,0 +1,119 @@
+from marionette_driver.by import By
+from marionette_driver.addons import Addons
+from marionette import MarionetteTestCase
+
+import os
+import sys
+import copy
+sys.path.insert(1, os.path.dirname(os.path.abspath(__file__)))
+
+from serversetup import LoopTestServers
+from config import FIREFOX_PREFERENCES
+from loopTestDriver import LoopTestDriver
+from config import CONTENT_SERVER_URL
+
+
+class Test2Linkclicker(LoopTestDriver, MarionetteTestCase):
+    # XXX Move this to setup class so it doesn't restart the server
+    # after every test.
+    def setUp(self):
+        # start server
+        self.loop_test_servers = LoopTestServers()
+
+        MarionetteTestCase.setUp(self)
+        LoopTestDriver.setUp(self, self.marionette)
+
+        standalone_url = CONTENT_SERVER_URL + "/"
+        prefs = copy.copy(FIREFOX_PREFERENCES)
+        prefs["loop.linkClicker.url"] = standalone_url
+
+        # Although some of these preferences might require restart, we don't
+        # use enforce_gecko_prefs (which would restart), as we need to restart
+        # for the add-on installation anyway.
+        self.marionette.set_prefs(prefs)
+
+        xpi_file = os.environ.get("LOOP_XPI_FILE")
+
+        if xpi_file:
+            addons = Addons(self.marionette)
+            # XXX We should really use temp=True here, but due to the later
+            # restart to ensure the add-on is installed correctly, this won't work
+            # at the moment. What we need is a fully restartless add-on - bug 1229352
+            # at which point we should be able to make this install temporarily
+            # and after the restart.
+            addons.install(os.path.abspath(xpi_file))
+
+        self.e10s_enabled = os.environ.get("TEST_E10S") == "1"
+
+        # Restart the browser nicely, so the preferences and add-on installation
+        # take full effect.
+        self.marionette.restart(in_app=True)
+
+        # this is browser chrome, kids, not the content window just yet
+        self.marionette.set_context("chrome")
+
+    def navigate_to_standalone(self, url):
+        self.switch_to_standalone()
+        self.marionette.navigate(url)
+
+    def standalone_check_view(self):
```

Comment:
I think this would be clearer if it was named as `standalone_check_own_link_view`